src/arch/src/aarch64/regs.rs

```diff
@@ -34,51 +34,38 @@ pub struct Aarch64Register {
 }
 
 /// Errors thrown while setting aarch64 registers.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum Error {
     /// Failed to get core register (PC, PSTATE or general purpose ones).
+    #[error("Failed to get {1} register: {0}")]
     GetCoreRegister(kvm_ioctls::Error, String),
-    /// Failed to get multiprocessor state.
-    GetMP(kvm_ioctls::Error),
-    /// Failed to get the register list.
-    GetRegList(kvm_ioctls::Error),
-    /// Failed to get a system register.
-    GetSysRegister(kvm_ioctls::Error),
-    /// A FamStructWrapper operation has failed.
-    Fam(utils::fam::Error),
     /// Failed to set core register (PC, PSTATE or general purpose ones).
+    #[error("Failed to set {1} register: {0}")]
     SetCoreRegister(kvm_ioctls::Error, String),
+    /// Failed to get a system register.
+    #[error("Failed to get register: {0}: {1}")]
+    GetRegister(u64, kvm_ioctls::Error),
+    /// Failed to set a system register.
+    #[error("Failed to set register {0}: {1}")]
+    SetRegister(u64, kvm_ioctls::Error),
+    /// Failed to get the register list.
+    #[error("Failed to retrieve list of registers: {0}")]
+    GetRegList(kvm_ioctls::Error),
+    /// Failed to get multiprocessor state.
+    #[error("Failed to get multiprocessor state: {0}")]
+    GetMP(kvm_ioctls::Error),
     /// Failed to Set multiprocessor state.
+    #[error("Failed to set multiprocessor state: {0}")]
     SetMP(kvm_ioctls::Error),
```

Comment:
nit: It might be better to use upper camel case on this occasion (like `Fam`, `GetMidrEl1`).
```suggestion
    GetMp(kvm_ioctls::Error),
    /// Failed to Set multiprocessor state.
    #[error("Failed to set multiprocessor state: {0}")]
    SetMp(kvm_ioctls::Error),
```

Reply:
done